pypicloud/cache/dynamo.py

```diff
@@ -22,6 +25,78 @@
 LOG = logging.getLogger(__name__)
 
 
+def _path_depth(package):
+    """Count the slashes in path, or return infinity on no path."""
+    path = package.data.get('path')
+    if not path:
+        return float('+inf')
+    return path.count('/')
+
+
+def _decide_between_versions(contender, current):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified < current.last_modified:
+        return contender
```

Comment:
Just realized that, for the fallout of the second case, if the files are equivalent, that won't send the operator on a goose chase. Since they're indiscernible, the operator will assume their actions got effected as expected. This means the expensive part of the consequences only happen when there's an actual mistake.

Now I'm even more set on that side! :P